l2geth/rollup/fees/rollup_fee.go

```diff
@@ -135,7 +140,7 @@ func PaysEnough(opts *PaysEnoughOpts) error {
 		overpaying := new(big.Int).Sub(opts.UserFee, opts.ExpectedFee)
 		threshold := mulByFloat(opts.ExpectedFee, opts.ThresholdUp)
 		// if overpaying > threshold, return error
-		if overpaying.Cmp(threshold) == 1 {
+		if overpaying.Cmp(threshold) == 1 && opts.UserFee.Cmp(previousFee) == 1 {
```

Comment:
Can we confirm the logic here? Sure we want &&?

Reply:
I think the logic here should be "are you overpaying according to both the current and previous limits?". Calculate "overpaying" as shown, calculate "overpaying_2" as Sub(opts.UserFee, opts.previousFee), calculate "threshold_2" by scaling up previousFee, then add "&& overpaying_2.cmp(threshold_2) == 1".

Reply:
See most recent commit - simplified the logic